### ISSUE TITLE
fix(link-fragments): slugPandoc strips periods and keeps leading non-letters

### ISSUE THREAD

**Issue by shinagawa-web** [shinagawa-web/gomarklint#217]
## Overview

`slugPandoc` (used by both `pandoc` and `quarto` presets) has two deviations from the Pandoc `auto_identifiers` specification.

## Problem

The [Pandoc manual](https://pandoc.org/MANUAL.html#extension-auto_identifiers) defines `auto_identifiers` as:

1. Remove all formatting, links, etc.
2. **Remove all punctuation, except underscores, hyphens, and periods.**
3. Replace all spaces and newlines with hyphens.
4. Convert all alphabetic characters to lowercase.
5. **Remove everything up to the first letter.**
6. If nothing is left after this, use `section`.

`slugPandoc` violates steps 2 and 5.

## Bug 1: Periods are stripped instead of kept

`slugPandoc` does not include `.` in its allowed character set.

| Input | Reference | `slugPandoc` |
|---|---|---|
| `Section 1.1` | `section-1.1` | `section-11` ❌ |
| `a.b.c` | `a.b.c` | `abc` ❌ |
| `1. Introduction` | `introduction` | `1-introduction` ❌ |

## Bug 2: Leading non-letters not stripped

`slugPandoc` does not implement step 5 ("remove everything up to the first letter").

| Input | Reference | `slugPandoc` |
|---|---|---|
| `123abc` | `abc` | `123abc` ❌ |
| `1. Introduction` | `introduction` | `1-introduction` ❌ |

## Fix

```go
func slugPandoc(text string) string {
    var sb strings.Builder
    sb.Grow(len(text))
    for _, r := range text {
        r = unicode.ToLower(r)
        if unicode.IsSpace(r) {
            sb.WriteByte('-')
        } else if (r >= 'a' && r <= 'z') || (r >= '0' && r <= '9') || r == '_' || r == '-' || r == '.' {
            sb.WriteRune(r)
        }
    }
    s := collapseDashes(sb.String())
    // step 5: remove everything up to the first letter
    for i, r := range s {
        if r >= 'a' && r <= 'z' {
            return s[i:]
        }
    }
    return ""
}
```

The `quarto` preset delegates to `slugPandoc` (Quarto uses `auto_identifiers` by default), so it is fixed automatically.

## References

- `internal/rule/link_fragments_slug.go` — `slugPandoc`
- `internal/rule/link_fragments_slug_test.go`
- Pandoc manual: https://pandoc.org/MANUAL.html#extension-auto_identifiers
- Verification comment: https://github.com/shinagawa-web/gomarklint/issues/206#issuecomment-4341846269
- Tracking issue: #206

**Comment by shinagawa-web** [comment]
## Implementation Plan

### Files to create
- none

### Files to modify
- `internal/rule/link_fragments_slug.go` — update `slugPandoc`: add `r == '.'` to allowed set; add leading-non-letter strip after `collapseDashes`
- `internal/rule/link_fragments_slug_test.go` — add test vectors: `"Section 1.1" → "section-1.1"`, `"123abc" → "abc"`, `"1. Introduction" → "introduction"`

### Checklist
- [ ] Fix `slugPandoc` in `link_fragments_slug.go` (periods kept + leading-letter strip)
- [ ] Update unit tests in `link_fragments_slug_test.go`
- [ ] Verify `quarto` preset is unaffected (it delegates to `slugPandoc`)
- [ ] `make test-all`
